pandas/core/tools/datetimes.py

```diff
@@ -369,15 +397,29 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
     if isinstance(arg, tslib.Timestamp):
         result = arg
     elif isinstance(arg, ABCSeries):
-        from pandas import Series
-        values = _convert_listlike(arg._values, True, format)
-        result = Series(values, index=arg.index, name=arg.name)
+        result = _maybe_convert_cache(arg, cache, tz)
+        if result is None:
+            from pandas import Series
+            values = _convert_listlike(arg._values, True, format)
+            result = Series(values, index=arg.index, name=arg.name)
     elif isinstance(arg, (ABCDataFrame, MutableMapping)):
         result = _assemble_from_unit_mappings(arg, errors=errors)
     elif isinstance(arg, ABCIndexClass):
-        result = _convert_listlike(arg, box, format, name=arg.name)
+        result = _maybe_convert_cache(arg, cache, tz)
+        if result is None:
+            result = _convert_listlike(arg, box, format, name=arg.name)
+        else:
```

Comment:
why can't you handle these caess list-list/index) *inside* ``_maybe_convert_cache``? (I am talking about the else/box part.